pkg/scene/widget_align_system.go

```diff
@@ -0,0 +1,200 @@
+package scene
+
+import (
+	"github.com/reearth/reearth-backend/pkg/id"
+)
+
+// WidgetAlignSystem is the layout structure of any enabled widgets that will be displayed over the scene.
+type WidgetAlignSystem struct {
+	inner WidgetZone
+	outer WidgetZone
+}
+
+// WidgetZone is the structure of each layer (inner and outer) of the align system.
+type WidgetZone struct {
+	left   WidgetSection
+	center WidgetSection
+	right  WidgetSection
+}
+
+// WidgetSection is the structure of each section of the align system.
+type WidgetSection struct {
+	top    WidgetArea
+	middle WidgetArea
+	bottom WidgetArea
+}
+
+// WidgetArea has the widgets and alignment information found in each part area of a section.
+type WidgetArea struct {
+	widgetIds []*id.WidgetID
+	align     string
+}
+
+type WidgetLocation struct {
+	Zone    string
+	Section string
+	Area    string
+}
+
+// NewWidgetAlignSystem returns a new widget align system.
+func NewWidgetAlignSystem() *WidgetAlignSystem {
+	return &WidgetAlignSystem{}
+}
+
+// WidgetZone will return a specific zone in the align system.
+func (was *WidgetAlignSystem) WidgetZone(zone string) *WidgetZone {
+	if was == nil {
+		return nil
+	}
+	switch zone {
+	case "inner":
+		return &was.inner
+	case "outer":
+		return &was.outer
+	}
+	return nil
+}
+
+// WidgetSection will return a specific section in the align system.
+func (was *WidgetAlignSystem) WidgetSection(zone, section string) *WidgetSection {
+	if was == nil {
+		return nil
+	}
+
+	z := was.WidgetZone(zone)
+
+	switch section {
+	case "left":
+		return &z.left
+	case "center":
+		return &z.center
+	case "right":
+		return &z.right
+	}
+	return nil
+}
+
+// WidgetArea will return a specific area in the align system.
+func (was *WidgetAlignSystem) WidgetArea(zone, section, area string) *WidgetArea {
+	if was == nil {
+		return nil
+	}
+
+	s := was.WidgetSection(zone, section)
+
+	switch area {
+	case "top":
+		return &s.top
+	case "middle":
+		return &s.middle
+	case "bottom":
+		return &s.bottom
+	}
+	return nil
+}
+
+// WidgetIds will return a slice of widget ids from a specific area.
+func (was *WidgetAlignSystem) WidgetIds(z, s, a string) []*id.WidgetID {
```

Comment:
```suggestion
func (was *WidgetAlignSystem) WidgetIDs(z, s, a string) []*id.WidgetID {
```

In Golang, `Id` in type names should be `ID`